source/core/TDCodecVia.cpp

```diff
@@ -1638,7 +1638,7 @@ void TDViaParser::ParseClump(VIClump* viClump, InstructionAllocator* cia)
     ClumpParseState state(viClump, cia, _pLog);
     SubString  token;
     SubString  instructionNameToken;
-    SubString  argExpressionTokens[ClumpParseState::kMaxArguments];
+    ExpandableStaticArray<SubString, ClumpParseState::kMaxArguments> argExpressionTokens;
```

Comment:
@spathiwa What do you think of using std::vector here instead of the new helper class?

Reply:
My reason for suggesting creating the class was to avoid the extra dynamic allocation in the common case where there are a limited number of arguments.  The idea of a class which uses static allocations for small/common situations but automatically expands to use dynamic allocation when necessary is a common design pattern and can be very beneficial.
However, in this case, where the same array is being reused every time to parse the arguments of every instruction, I concede that adding a reserve() call would get most of the benefits of the class.  However, since Jared did the work of making the class on my recommendation, and it's a judgement call whether it's a win in this case but could be useful in other areas, I say he should keep it.

Reply:
I concur with @sanmut, please use a std::vector instead and reserve ClumpParseState:kMaxArguments for it. That will make it a smaller fix, which is great because we want to patch this if there is another patch for NXG 3.1.